flocker/control/_model.py

```diff
@@ -930,6 +950,38 @@ def get_information_wipe(self):
         return _WipeNodeState(node_uuid=self.uuid, attributes=attributes)
 
 
+@implementer(IClusterStateChange)
+class UpdateNodeStateEra(PClass):
+    """
+    Update a node's era.
+
+    :ivar UUID uuid: The node's UUID.
+    :ivar UUID era: The node's era.
+    """
+    uuid = field(type=UUID, mandatory=True)
+    era = field(type=UUID, mandatory=True)
+
+    def update_cluster_state(self, cluster_state):
+        """
+        Record the node's era and discard the ``NodeState`` if it doesn't
+        match the era.
+        """
+        if cluster_state.node_uuid_to_era.get(self.uuid) != self.era:
+            # Discard the NodeState:
```

Comment:
:grey_question: Question on future work: With this here, do we still have use for the wiping mechanism? it seems that this will let us remove that. (Also, if we can get rid of wiping a big +1 to that idea, as a newbie it always takes me awhile to wrap my head around why we need wiping, and it's very confusing that we only implement it for the 1 specific state change that needs it).